Port cloud datasets page updates to enterprise 26.1 (#1437)
seqeralabs/docs: platform-enterprise_docs/data/datasets.md

:::

Datasets are CSV (comma-separated values) and TSV (tab-separated values) files stored in, or linked to, a workspace. They are used as inputs to pipelines to simplify data management, minimize user data-input errors, and facilitate reproducible workflows.

On the datasets screen, you can:

- Upload directly or link to an externally hosted dataset.
- View the count of pipeline runs in the workspace that have used a specific dataset input.
- Apply multiple labels to datasets for easier searching and grouping.
- Sort datasets by name, most recently updated, and most recently used.
- Hide datasets that are not used in the workspace.
- View dataset metadata (created by, last updated, last used).
- Edit dataset details (name, description, and labels).
- Create new versions of an uploaded dataset.

## Benefits

- Datasets reduce errors that occur due to manual data entry when you launch pipelines.
- Datasets can be generated automatically in response to events (such as S3 storage new file notifications).
- Datasets can streamline differential data analysis when using the same pipeline to launch a run for each dataset as it becomes available.

## Format

> The most commonly used datasets for Nextflow pipelines are samplesheets, where each row consists of a sample, the location of files for that sample (such as FASTQ files), and other sample details. For example, [*nf-core/rnaseq*](https://github.com/nf-core/rnaseq) works with input datasets (samplesheets) containing sample names, FASTQ file locations, and indications of strandedness. The Seqera Community Showcase sample dataset for *nf-core/rnaseq* looks like this:

Reviewer: ```suggestion
The most commonly used datasets for Nextflow pipelines are sample sheets, where each row consists of a sample identifier, the location of the sample's files (e.g., FASTQ files), and other sample details. For example, [*nf-core/rnaseq*](https://github.com/nf-core/rnaseq) works with input datasets (sample sheets) that include sample names, FASTQ file locations, and strandedness annotations. The Seqera Community Showcase sample dataset for *nf-core/rnaseq* looks like this:
```

For your pipeline to use your dataset as input during runtime, information about the dataset and file format must be included in the relevant parameters of your [pipeline schema](../pipeline-schema/overview). The pipeline schema specifies the accepted dataset file type in the `mimetype` attribute (either `text/csv` or `text/tsv`).

## Dataset file content requirements and validation

Datasets can point to files stored in various locations, such as Amazon S3, GitHub, or Hugging Face. To stage the file paths defined in the dataset, Nextflow requires access to the infrastructure where the files reside, whether on cloud or HPC systems. Add the access keys for data sources that require authentication to your [secrets](../secrets/overview).

:::note
Seqera doesn't validate your dataset file contents. While datasets can contain static file links, you're responsible for maintaining the access to that data.
:::

## Add a dataset

All Seqera user roles have access to the datasets feature in organization workspaces. There are two ways to add a dataset:

1. Direct upload: This is the best option if immutability is a requirement but size cannot exceed 10 MB.

> 1. Link to an externally hosted file: This is the best option for very large size files but relies on the external hosting service for availability and immutability.

Reviewer 2: ```suggestion
1. Direct upload: This is the best option if immutability is a requirement but size cannot exceed 10 MB.
2. Link to an externally hosted file: This is the best option for very large size files but relies on the external hosting service for availability and immutability.
```

### Direct upload

1. In the sidebar navigation, select **Datasets**.
1. Select **Add Dataset** and choose **Upload file**.
1. Complete the **Name** and **Description** fields using information relevant to your dataset.
1. Optionally add one or more **Labels** to your dataset. You can use labels as a search filter but they don't apply to other resources in Seqera.
1. Upload a dataset to your workspace with drag-and-drop or use the **Upload file** file explorer dialog.
1. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.

> 1. Select **Add**.

Reviewer 2: ```suggestion
1. In the sidebar navigation, select **Datasets**.
2. Select **Add Dataset** and choose **Upload file**.
3. Complete the **Name** and **Description** fields using information relevant to your dataset.
4. Optionally add one or more **Labels** to your dataset. You can use labels as a search filter but they don't apply to other resources in Seqera.
5. Upload a dataset to your workspace with drag-and-drop or use the **Upload file** file explorer dialog.
6. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.
7. Select **Add**.
```

:::warning
The size of the uploaded dataset file cannot exceed 10 MB.
:::

### Link to an externally hosted file

1. In the sidebar navigation, select **Datasets**.
1. Select **Add Dataset** and choose **Link to URL**.
1. Complete the **Name** and **Description** fields using information relevant to your dataset.
1. Optionally add one or more **Labels** to your dataset. You can use labels as a search filter but they don't apply to other resources in Seqera.
1. Copy and paste the dataset URL into the **Dataset URL** field.
1. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.
1. Select **Add**.

> 1. The dataset will be displayed with a `Linked` badge for easy identification.

Reviewer 2: ```suggestion
1. In the sidebar navigation, select **Datasets**.
2. Select **Add Dataset** and choose **Link to URL**.
3. Complete the **Name** and **Description** fields using information relevant to your dataset.
4. Optionally add one or more **Labels** to your dataset. You can use labels as a search filter but they don't apply to other resources in Seqera.
5. Copy and paste the dataset URL into the **Dataset URL** field.
6. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.
7. Select **Add**.
8. The dataset will be displayed with a `Linked` badge for easy identification.
```

## Manage dataset versions

For directly uploaded datasets, Seqera can manage multiple versions.

:::note
For linked datasets, versioning is unavailable.
:::

### Add a dataset version

1. Select the three dots next to the dataset you want to add a new version for.
1. Select **Add version**.
2. Upload a dataset to your workspace with drag-and-drop or use the system **Upload file** file explorer dialog.
3. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.

> 4. Select **Add**.

Reviewer 2: ```suggestion
2. Select **Add version**.
3. Upload a dataset to your workspace with drag-and-drop or use the system **Upload file** file explorer dialog.
4. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.
5. Select **Add**.
```
@justinegeffen sorry my previous suggestion was incorrect. Can we apply this too please? Thanks

> 1. Select **Add**.

Reviewer 2: ```suggestion
1. Select **Add version**.
2. Upload a dataset to your workspace with drag-and-drop or use the system **Upload file** file explorer dialog.
3. For datasets that use their first row for column names, customize the dataset view using the **Set first row as header** option.
4. Select **Add**.
```

Once disabled, a dataset version cannot be re-enabled.
:::

## Use a dataset

To use a dataset with pipelines added to your workspace:

1. Open any pipeline that contains a pipeline schema from the [Launchpad](../launch/launchpad).
1. Select the input field for the pipeline, removing any default values.

> 1. Pick the dataset to use as input to your pipeline.

Reviewer 2: ```suggestion
2. Select the input field for the pipeline, removing any default values.
3. Pick the dataset to use as input to your pipeline.
```
